jetty-core/jetty-server/src/main/config/modules/delay-until-content.mod

```diff
@@ -18,3 +18,8 @@ threadlimit
 [xml]
 etc/jetty-delayed.xml
 
+
+[ini-template]
+#tag::documentation[]
+## The maximum bytes to retain whilst delaying content; or 0 for no delay; or -1 (default) for a default value.
```

Comment:
We often use -1 for a heuristic or default value.   I strongly think that the default value of one input buffer is a good value as it represents no additional memory commitment, so no thought is needed here.

Reply:
@gregw "one input buffer" is confusing.

An application would be delayed, and then it would read a chunk that is _not_ "one input buffer" in size, because some bytes are stolen by chunking or by HTTP/2 framing.

Some other times the input buffer will contain both headers and content, so it would be even less.

The value `1` has exactly the same semantic: as soon as there is at least one byte of content, we call the application, with exactly the same memory commitment (that byte has been read in an input buffer), so with your proposal we would have 2 values that indicate the same thing, and we don't have a value to indicate "all content".

Reply:
@sbordet The use doesn't need to know it is one input buffer.  As far as the use knows it is just a default heuristic value.  But note that the implementation does  take into account framing overheads.     

The value 1 doesn't have the same semantic, as that will invoke the application on the first chunk, potentially with a large empty buffer. My -1 will delay the calling the application until EOF or maximal usage of the allocated buffer.

With your proposal, -1 and MAX_VALUE have the same semantic.